app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabsIntegration.kt

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.tabstray.syncedtabs
+
+import android.content.Context
+import androidx.lifecycle.LifecycleOwner
+import androidx.navigation.NavController
+import mozilla.components.browser.storage.sync.SyncedDeviceTabs
+import mozilla.components.feature.syncedtabs.SyncedTabsFeature
+import mozilla.components.feature.syncedtabs.storage.SyncedTabsStorage
+import mozilla.components.feature.syncedtabs.view.SyncedTabsView
+import mozilla.components.service.fxa.manager.FxaAccountManager
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.base.observer.Observable
+import mozilla.components.support.base.observer.ObserverRegistry
+import org.mozilla.fenix.tabstray.TabsTrayAction
+import org.mozilla.fenix.tabstray.TabsTrayStore
+
+class SyncedTabsIntegration(
+    private val store: TabsTrayStore,
+    private val context: Context,
+    private val navController: NavController,
+    storage: SyncedTabsStorage,
+    accountManager: FxaAccountManager,
+    lifecycleOwner: LifecycleOwner,
+) : LifecycleAwareFeature,
+    SyncedTabsView,
+    Observable<SyncedTabsView.Listener> by ObserverRegistry() {
+
+    private val syncedTabsFeature by lazy {
+        SyncedTabsFeature(
+            context = context,
+            storage = storage,
+            accountManager = accountManager,
+            view = this,
+            lifecycleOwner = lifecycleOwner,
+            onTabClicked = {
+                // We can ignore this callback here because we're not connecting the adapter
+                // back to the feature. This works fine in other features, but passing the listener
+                // to other components in this case is annoying.
+            }
+        )
+    }
+
+    override var listener: SyncedTabsView.Listener? = null
+
+    private val syncButtonBinding by lazy {
+        SyncButtonBinding(store) { listener?.onRefresh() }
+    }
+
+    override fun start() {
+        syncedTabsFeature.start()
+        syncButtonBinding.start()
+    }
+
+    override fun stop() {
+        syncedTabsFeature.stop()
+        syncButtonBinding.stop()
+    }
+
+    override fun onError(error: SyncedTabsView.ErrorType) {
```

Comment:
@jonalmeida wanted to double check that it's okay with dropping the `scope.launch {}` seen in https://github.com/mozilla-mobile/fenix/pull/22798/files#diff-1fd4cbb372da638657848eed86633cb5344dceb48f729a2eca27e230bbcdd593L85. It doesn't look like we need it, but wanted to check about the previous context.

Reply:
Good catch. I don't remember if I tested out the error case, this might end up failing to do anything if we aren't invoking this on the main thread.

The quickest way to test this is to connect a device that has tabs here. Then on desktop, go to the account settings and disconnect your test device. Cold boot Fenix and that should put you in an error state.

Reply:
I was able to verify one of the error scenarios by turning off WiFi and cold booting the app. 
![image](https://user-images.githubusercontent.com/87384386/149565290-feae279d-9e84-4172-9543-c859ea8443d9.png)

Reply:
@jonalmeida I am not too familiar with the original design spec. Do we have a Figma that tells us what each of the state should look like?

Reply:
No. All the errors should look the same as the screenshot above to match these error types: https://github.com/mozilla-mobile/android-components/blob/main/components/feature/syncedtabs/src/main/java/mozilla/components/feature/syncedtabs/view/SyncedTabsView.kt#L61-L88

Reply:
You can look at the code to know what it has to look like too.

Reply:
Four of the errors are just text. One is text and a button.

Reply:
This bit connects the feature to the UI and tells it when to show the tab list or when to show an error: https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabsTrayLayout.kt